hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/validator/JsonTopicMessageValidator.java

```diff
@@ -35,7 +38,12 @@ public void check(byte[] message, Topic topic) {
         List<String> errors = validate(schemaRepository.getSchema(topic), message);
 
         if (!errors.isEmpty()) {
-            throw new InvalidMessageException("Message incompatible with JSON schema", errors);
+            if (topic.isValidationDryRunEnabled()) {
+                logger.info("Message incompatible with JSON schema for topic {}, errors: {}, message body: {}",
+                        topic.getQualifiedName(), Joiner.on(";").join(errors), new String(message, UTF_8));
```

Comment:
For this experimental dry run, we would like to be able to investigate _why_ it didn't pass the validation.